policies/optimized-images.md

```diff
@@ -0,0 +1,342 @@
+# Optimized Image Policies Explainer
+
+loonybear@, last updated: 4/25/2018
+
+<span style="color:#38761d;">Status: Draft Proposal</span>
+
+
+## Goal
+
+Images make up majority of the downloaded bytes on websites. In addition, images often occupy a significant amount of visual space. Optimizing images can improve loading performance and reduce network traffic. Surprisingly, more than half of the sites on the web, including advanced ones, are shipping unoptimized images. This means these sites can all achieve a performance improvement by serving optimized images.
+
+Optimized image policies are aiming to solve problems with sites shipping images that are poorly compressed or unnecessarily large.
+
+
+## What are "Optimized Image Policies"?
+
+Optimized image policies introduce a set of restrictions (policies) on images that can be applied with dev-time enforcement. An image will be rendered with **inverted colors** when violating a policy, making it easy for web developers to identify and fix the error.
+
+
+### Optimized image policies
+
+*   **["legacy-image-formats" policy](#legacy-image-formats)**
+    *   Images must be of one of the _**modern formats***_ (JPEG, PNG, WEBP, etc).
+*   **["maximum-downscaling-image" policy](#maximum-downscaling-image)**
+    *   Images must not be bigger than its container size by more than _**X times***_ .
+*   **["image-compression" policy](#image-compression)**
+    *   Images used in rendering must not include too much metadata.
+    *   Images must not be more than _**X bits***_ per compressed pixel.
+
+**Note**: We want to allow developers the ability to make the final decision about the tradeoffs they make. * means developers will eventrually be able to specify the "value" of the policy. For example, `maximum-downscaling-image(2)` specifies the maximum ratio (2) images are allowed to be downscaled by.
+
+
+## Detailed policy discussion
+
+<a name="legacy-image-formats">
+
+### "legacy-image-formats" policy
+
+</a>
+
+Image formats affect file size, loading speed and appearance. Modern image formats yield large byte savings and performance improvement. `legacy-image-formats` is a policy controlled feature that restricts images to be one of certain modern formats.
+
+When a document is disallowed to use `legacy-image-formats` policy, its `<img>` elements will render images of "legacy" formats with inverted colors.
+
+
+#### Specification
+
+- The default list of modern image formats is: JPEG, PNG, GIF, WEBP, and SVG.
```

Comment:
GIF, really? When's the right time to use GIF on the web? If anything, seems like we should be pushing folks aggressively away from GIF to video.